scripts/agent_step_zkai_stwo_composition.py

```
@@ -0,0 +1,594 @@
+
+def _source_evidence_path(path: pathlib.Path) -> pathlib.Path:
+    resolved = path.resolve()
+    root = ROOT.resolve()
+    if resolved != root and root not in resolved.parents:
+        raise CompositionError(f"source evidence path escapes repository: {path}")
+    if not resolved.is_file():
+        raise CompositionError(f"source evidence path is missing: {path}")
+    return resolved
+
+
+def _validate_stwo_evidence_payload(payload: Any) -> dict[str, Any]:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `payload`

(ANN401)